lotus/sem_ops/sem_extract.py

```diff
@@ -42,41 +46,41 @@ def sem_extract(
     postprocess_output = postprocessor(lm_output.outputs)
     lotus.logger.debug(f"raw_outputs: {lm_output.outputs}")
     lotus.logger.debug(f"outputs: {postprocess_output.outputs}")
-    lotus.logger.debug(f"quotes: {postprocess_output.quotes}")
 
     return SemanticExtractOutput(**postprocess_output.model_dump())
 
 
 @pd.api.extensions.register_dataframe_accessor("sem_extract")
-class SemExtractDataframe:
-    """DataFrame accessor for semantic extract."""
-
-    def __init__(self, pandas_obj: Any):
+class SemExtractDataFrame:
+    def __init__(self, pandas_obj: pd.DataFrame):
         self._validate(pandas_obj)
         self._obj = pandas_obj
 
     @staticmethod
-    def _validate(obj: Any) -> None:
+    def _validate(obj: pd.DataFrame) -> None:
         if not isinstance(obj, pd.DataFrame):
             raise AttributeError("Must be a DataFrame")
 
     def __call__(
         self,
+        columns: list[str],
         user_instruction: str,
```

Comment:
I think these argument names are a bit confusing --
I don't think we really need a user_instruction, if it's always just a column wrapped in a string, we should just rename it input_col, or input_cols (so we can take a list). Is there any case when we actually need a string parameterized by the columns? if so, we can keep the parameter name, but move user_instruction to be the first parameter (to keep things consistent with the other sem ops). Otherwise, we should rename user_instruction to input_cols, and columns to output_cols

Reply:
Agree

Reply:
I also think it would give us more flexibility if columns (really output_schema) is type Dict[str, str | None] -- that way the user can specify a description for each field if they want, or not and just use the default None

Reply:
Changed to `input_cols`, `output_cols`